java/ql/src/Security/CWE/CWE-312/AllowBackupAttributeEnabled.qhelp

```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>The Android manifest file defines configuration settings for Android 
+applications. In this file, the <code>android:allowBackup</code> attribute of
+the <code>application</code> element can be used to define whether or not the
+application can have automatic backups.</p>
+
+<p>Enabling backups may allow an attacker to extract sensitive data. Therefore,
+it is advised to set <code>android:allowBackup</code> to <code>false</code> if 
+your application uses any sensitive data.</p>
+</overview>
+
+<recommendation>
+<p>For Android applications which process sensitive data, set the 
+<code>android:allowBackup</code> setting to <code>false</code> in the manifest
+file.</p>
+
+<p>Note: Since Android 6.0 (Marshmallow), applications default to participating
+in automatic backups. Therefore, it is necessary to explicitly disable backups.
+</p>
+</recommendation>
+
+<example>
+
+<p>In the two examples below, the <code>android:allowBackup</code> setting is enabled:</p>
```

Comment:
```suggestion
<p>In the following two (bad) examples, the <code>android:allowBackup</code> setting is enabled:</p>
```

Slight edit in line with [MS style guide](https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/b/below).